lib/pbench/cli/server/tree_manage.py

```diff
@@ -163,7 +169,7 @@ def tree_manage(context: object, display: bool, lifetime: int, reclaim: bool):
         if display:
             print_tree(cache_m)
         if reclaim:
-            reclaim_cache(cache_m, logger, lifetime)
+            reclaim_cache(cache_m, logger, reclaim)
```

Comment:
Your instinct isn't necessarily incorrect, however if *I* were combining them I'd want to see which datasets were reclaimed, not just a static display of the state *after* reclaim. Printing the tree first would allow you to examine the timestamps. A "+one" might be to pass the `reclaim` timestamp into `print_tree` and have it add an annotation showing explicitly which *will* be reclaimed: but that seemed excessive.

Reply:
Clearly, one could run the command with just `--print_tree` to see the existing tree, and then run it with just `--reclaim` to do the reclamation, and then run it again with just `--print_tree` to see the updated state (and, we have the logger output and the audits which will report what the changes were...), and then there would be no ambiguities.

However, if the user specifies both `--print_tree` and `--reclaim`, then the question arises as to what state the user _expects_ to be displayed.  I'm not sure that there is any point to displaying the current state just before changing it.  (If the user really wants to see the status quo ante, s/he should omit the `--reclaim` option...but, I suppose it's equally valid to say that, if the user really wants to see the status quo post, s/he should run the command again with just `--print_tree` afterwards.)

Nevertheless, what I'm trying to do is to avoid surprise.  Perhaps what we should only accept one of the two options and either ignore the other or report an error if they are both provided?

Reply:
This is a fairly specialized admin command. I really don't see any problem, or any point in worrying about it.